src/file_storage.cpp

```diff
@@ -802,7 +802,7 @@ namespace libtorrent
 	std::int64_t file_storage::file_size(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return std::int64_t(m_files[std::size_t(index)].size);
+		return m_files[std::size_t(index)].size;
```

Comment:
I didn't find a specific flag for this (it's very particular), the current flag `-Wno-sign-conversion` suppress many more (sometimes suspicions) conversions. I just finished my branch, no way I can send a PR from it, I will send first the headers and later the srcs in groups.